device/core/src/internal_client.ts

```diff
@@ -337,6 +339,68 @@ export abstract class InternalClient extends EventEmitter {
     }, abandonCallback);
   }
 
+  /**
+   * Sends a property update patch to the Azure IoT Hub or Azure IoT Edge Hub service.
+   *
+   * @param {ClientPropertyCollection} propertyCollection A ClientPropertyCollection object representing the property patch.
+   * @param {(err: Error) => void}     done               A callback used to get notified of the success or failure of the operation.
+   *                                                      If no callback is specified, a promise is returned instead.
+   */
+  updateClientProperties(propertyCollection: ClientPropertyCollection): Promise<void>;
+  updateClientProperties(propertyCollection: ClientPropertyCollection, done: (err: Error) => void): void;
+  updateClientProperties(propertyCollection: ClientPropertyCollection, done?: (err: Error) => void): Promise<void> | void {
+    return callbackToPromise((_callback) => {
+      this.getTwin((err, twin) => {
+        if (err) {
+          _callback(err);
+          return;
+        }
+        (twin as any)._updateReportedProperties(propertyCollection.backingObject, _callback);
+      });
+    }, done);
+  }
+
+  /**
+   * Registers a listener to get invoked with a writable property patch whenever the device receives a writable property request.
+   *
+   * @param {(properties: ClientPropertyCollection) => void} callback The callback to get invoked on a writable property request.
+   */
+  onWritablePropertyUpdateRequest(callback: (properties: ClientPropertyCollection) => void): void {
+    this.getTwin((err, twin) => {
+      if (err) {
+        this.emit('error', err);
+        return;
+      }
+      twin.on('properties.desired', (patch) => {
+        callback(new ClientPropertyCollection(patch));
+      });
+    });
+  }
+
+  /**
+   * Gets the client properties the Azure IoT Hub or Azure IoT Edge Hub service.
+   * This method is only intended for use with Azure IoT Plug and Play.
+   *
+   * @param {Callback<ClientProperties>} done The callback which gets invoked with the ClientProperties object.
+   *                                          If no callback is specified, a promise is returned instead.
+   */
+  getClientProperties(): Promise<ClientProperties>;
+  getClientProperties(done: Callback<ClientProperties>): void;
+  getClientProperties(done?: Callback<ClientProperties>): Promise<ClientProperties> | void {
+    return callbackToPromise((_callback) => {
+      this.getTwin((err, twin) => {
+        if (err) {
+          _callback(err);
+          return;
+        }
+        if (typeof twin.properties.reported.update === 'function') {
```

Comment:
I think I know what you are doing here but you gotta explain this.

Reply:
Added a comment to explain this (a whole paragraph was needed to fully explain)